src/libs/actions/PersonalDetails/index.native.js

```diff
@@ -0,0 +1,329 @@
+import * as API from '../../API';
+import NameValuePair from '../NameValuePair';
+
+let currentUserEmail = '';
+Onyx.connect({
+    key: ONYXKEYS.SESSION,
+    callback: val => currentUserEmail = val ? val.email : '',
+});
+
+let personalDetails;
+Onyx.connect({
+    key: ONYXKEYS.PERSONAL_DETAILS,
+    callback: val => personalDetails = val,
+});
+
+/**
+ * Helper method to return a default avatar
+ *
+ * @param {String} [login]
+ * @returns {String}
+ */
+function getDefaultAvatar(login = '') {
+    // There are 8 possible default avatars, so we choose which one this user has based
+    // on a simple hash of their login (which is converted from HEX to INT)
+    const loginHashBucket = (parseInt(md5(login).substring(0, 4), 16) % 8) + 1;
+    return `${CONST.CLOUDFRONT_URL}/images/avatars/avatar_${loginHashBucket}.png`;
+}
+
+/**
+ * Returns the URL for a user's avatar and handles someone not having any avatar at all
+ *
+ * @param {Object} personalDetail
+ * @param {String} login
+ * @returns {String}
+ */
+function getAvatar(personalDetail, login) {
+    if (personalDetail && personalDetail.avatarThumbnail) {
+        return personalDetail.avatarThumbnail;
+    }
+
+    return getDefaultAvatar(login);
+}
+
+/**
+ * Returns the displayName for a user
+ *
+ * @param {String} login
+ * @param {Object} [personalDetail]
+ * @returns {String}
+ */
+function getDisplayName(login, personalDetail) {
+    // If we have a number like +15857527441@expensify.sms then let's remove @expensify.sms
+    // so that the option looks cleaner in our UI.
+    const userLogin = Str.removeSMSDomain(login);
+    const userDetails = personalDetail || personalDetails[login];
+
+    if (!userDetails) {
+        return userLogin;
+    }
+
+    const firstName = userDetails.firstName || '';
+    const lastName = userDetails.lastName || '';
+
+    return (`${firstName} ${lastName}`).trim() || userLogin;
+}
+
+/**
+ * Format personal details
+ *
+ * @param {Object} personalDetailsList
+ * @return {Object}
+ */
+function formatPersonalDetails(personalDetailsList) {
+    return _.reduce(personalDetailsList, (finalObject, personalDetailsResponse, login) => {
+        // Form the details into something that has all the data in an easy to use format.
+        const avatar = getAvatar(personalDetailsResponse, login);
+        const displayName = getDisplayName(login, personalDetailsResponse);
+        const pronouns = lodashGet(personalDetailsResponse, 'pronouns', '');
+        const timezone = lodashGet(personalDetailsResponse, 'timeZone', CONST.DEFAULT_TIME_ZONE);
+
+        return {
+            ...finalObject,
+            [login]: {
+                login,
+                avatar,
+                displayName,
+                pronouns,
+                timezone,
+            },
+        };
+    }, {});
+}
+
+/**
+ * Get the personal details for our organization
+ */
+function fetch() {
+    API.Get({
+        returnValueList: 'personalDetailsList',
+    })
```

Comment:
NAB: I've seen this a few times where you're putting object definitions with a single field onto multiple lines, but IMO it's cleaner in this case to just put everything on a single line.

```suggestion
    API.Get({ returnValueList: 'personalDetailsList' })
```

Reply:
I think this is something that was in the previous version itself. I did not change this function.